1-Draft/RFCXXXX-Obsoletion-and-Deprecation.md

```diff
@@ -0,0 +1,511 @@
+
+This section provides proposals on global behaviors as well as overriding these behaviors.
+
+### Displaying obsolete behavior
+
+Currently, all warnings are displayed based on the global preference variable `WarningPreference`.
+Additional levels of granularity would be useful so different warnings may be controlled, much the same way that trap/catch works in PowerShell today.
+While it is the case that you can use `-WarningAction SilentlyContinue` with cmdlets today to suppress obsolescence warnings, this will affect _all_ warnings.
+In some cases, you may not want to suppress obsolete warnings while suppressing operational warnings (and vice-versa).
+
+#### ObsoleteWarningPreference
+
+This is a new ubiquitous variable will dictate the global behavior of PowerShell when an item has been made obsolete.
+This has four possible values:
+
+- SilentlyContinue
+  - Warning messages will not be shown
+- Continue
+  - Warning messages will be shown
+- NotAvailable
+  - Any obsolete item will be not found.
+   This means that cmdlets will produce a CommandNotFound error and types (or their obsolete members) will produce a runtime error.
+   Additionally, the warning message will not be produced.
+- Stop
+  - Warning messages will be converted to terminating error messages
+
+---
+**NOTE**
+A ubiquitous parameter could be added, but this seems less interesting as the user must have _a priori_ knowledge about the obsolete cmdlet, thus diluting its usefulness.
+It may be possible to use a ubiquitous parameter via `PSDefaultParameterValues`, but this feels like _additional_ complexity rather than simplification.
+Additionally, it would not be applicable to the use of obsolete _types_, further reducing its usefulness.
+
+---
+
+### Overriding Behaviors
+
+It is not enough to provide default behaviors, as There are conditions where the default is not desired, so it must be possible to locally change PowerShell behaviors.
+In order to support this scenario an additional cmdlet `Set-ObsoleteWarning` shall be provided.
+This allows the user to specify for a specific warning what behavior is to be provided.
+
+`Set-ObsoleteWarning -Id _WarningId_ -Action _PreferenceValue_`
+
+The _WarningId_ may be retrieved with the `Get-ObsoleteItem` cmdlets.
+
+`Get-ObsoleteWarning` will return the current overrides in place.
+
+Enables individual tuning for a specific obsolete warning.
+
+Detailed discussion on these tools is [below](#configuration-tools).
+
+### Discovery
+
+Being able to discover and explore the environment is one of PowerShells' great strengths.
+We should provide tools which enable users to discover obsolete cmdlets and types in their environment.
+PowerShell currently supports the attribute `ObsoleteAttribute` which should be sufficient.
+
+## Additions to the module manifest
+
+In order to quickly report on whether a module has some element which is obsolete, the module manifest will be extended to support providing a hint to obsolescence.
```

Comment:
The risk of advertising obsolete data in the module manifest is requiring the manifest to match the code.  Which becomes the source of truth?  Perhaps just relying on the code unless there is a big perf need to just rely on the manifest.